{
    "MITRE": {
        "CVE_data_meta": {
            "ASSIGNER": "security@apache.org",
            "ID": "CVE-2021-35940",
            "STATE": "PUBLIC",
            "TITLE": "Regression of CVE-2017-12613"
        },
        "affects": {
            "vendor": {
                "vendor_data": [
                    {
                        "product": {
                            "product_data": [
                                {
                                    "product_name": "Apache Portable Runtime (APR)",
                                    "version": {
                                        "version_data": [
                                            {
                                                "version_affected": "=",
                                                "version_name": "Apache Portable Runtime",
                                                "version_value": "1.7.0"
                                            }
                                        ]
                                    }
                                }
                            ]
                        },
                        "vendor_name": "Apache Software Foundation"
                    }
                ]
            }
        },
        "credit": [
            {
                "lang": "eng",
                "value": "The Apache Portable Runtime project would like to thank Iveta Cesalova (Red Hat) for reporting this issue."
            }
        ],
        "data_format": "MITRE",
        "data_type": "CVE",
        "data_version": "4.0",
        "description": {
            "description_data": [
                {
                    "lang": "eng",
                    "value": "An out-of-bounds array read in the apr_time_exp*() functions was fixed in the Apache Portable Runtime 1.6.3 release (CVE-2017-12613).  The fix for this issue was not carried forward to the APR 1.7.x branch, and hence version 1.7.0 regressed compared to 1.6.3 and is vulnerable to the same issue."
                }
            ]
        },
        "generator": {
            "engine": "Vulnogram 0.0.9"
        },
        "impact": [
            {}
        ],
        "problemtype": {
            "problemtype_data": [
                {
                    "description": [
                        {
                            "lang": "eng",
                            "value": "APR apr_time_exp* out of bounds array dereference"
                        }
                    ]
                }
            ]
        },
        "references": {
            "reference_data": [
                {
                    "refsource": "CONFIRM",
                    "url": "http://svn.apache.org/viewvc?view=revision&revision=1891198"
                },
                {
                    "refsource": "CONFIRM",
                    "url": "http://mail-archives.apache.org/mod_mbox/www-announce/201710.mbox/%3CCACsi251B8UaLvM-rrH9fv57-zWi0zhyF3275_jPg1a9VEVVoxw@mail.gmail.com%3E"
                },
                {
                    "refsource": "CONFIRM",
                    "url": "https://dist.apache.org/repos/dist/release/apr/patches/apr-1.7.0-CVE-2021-35940.patch"
                },
                {
                    "refsource": "CONFIRM",
                    "url": "https://lists.apache.org/thread.html/ra2868b53339a6af65577146ad87016368c138388b09bff9d2860f50e%40%3Cdev.apr.apache.org%3E"
                }
            ]
        },
        "source": {
            "discovery": "UNKNOWN"
        }
    }
}